### DISCUSSION TITLE
Scripting on gopro hero 7

### DISCUSSION THREAD

**Discussion by softyoda** [gopro/labs#762] [Q&A]
Hi, as stated in my comment [here ](https://github.com/gopro/labs/issues/751#issuecomment-1934603550), I wish to know if it would be possible to do some basic script on the GoPro Hero 7 Black without having to upgrade my entire fleet of GoPros.
I wish to shoot synced multi-bracket timelapse of photos each 30sec.

The program would be something among those lines: 
```
-wait until the timecode enter to new minute
-while : no-stopped
    -take one photo with 0EV bracketing and save it as [gopro-name][mid].jpg
    -take one photo with -2EV bracketing and save it as [gopro-name][shadow].jpg
    -take one photo with +2EV bracketing and save it as [gopro-name][highlights].jpg
    -wait 30sec
```

Would it be possible with Lab, if not is there any other ways I could script that ?
Thanks in advance for your answers.

**Comment by dnewman-gpsw** [comment]
While you can change the filename, try first without.

The best you can do on HERO7 is jamsync the time with https://gopro.github.io/labs/control/precisiontime/

Then crudely run a command like
dP!10:00Nx0!Sx-2!Sx2!S!10:01Nx0!Sx-2!Sx2!S!10:02Nx0!Sx-2!Sx2!S!10:03Nx0!Sx-2!Sx2!S

Scripting every frame, for a particular time, but it works.  You could extend this crazy mess with dozens of scripts stored on the SD card.

Much easier on HERO11 & 12 which has variables and more time event conditionals.
